### PR TITLE
Add rpmsgmtd support

### PR DESCRIPTION
**Pull request by CV-Bowen**
## Summary
1. move the find_mtddriver() to fs.h and add close_mtddriver() api, because the rpmsgmtd need  find_mtddriver() to get the mtd device according to the mtd device path, and need close_mtddriver() to release the inode;
2. add rpmsgmtd support, the local cpu can access the remote cpu mtd device;

## Impact
New feature, should be no impact

## Testing
test with sim:rpserver and sim:rpproxy

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: drivers/mtd/rpmsgmtd.c [hunk 1]
```diff
@@ -0,0 +1,1135 @@
...
+  struct rpmsgmtd_erase_s msg;
+  int ret;
+
+  /* Sanity checks */
+
+  DEBUGASSERT(priv != NULL);
+
+  msg.startblock = startblock;
+  msg.nblocks    = nblocks;
+  ret = rpmsgmtd_send_recv(priv, RPMSGMTD_ERASE, true, &msg.header,
```
Comment (anchao) [inline drivers/mtd/rpmsgmtd.c:176]: remove ret

File: drivers/mtd/Kconfig [hunk 2]
```diff
@@ -345,6 +345,16 @@ config NULLMTD_ERASESTATE
 
 endif # NULLMTD
 
+config RPMSGMTD
```
Comment (anchao) [inline drivers/mtd/Kconfig:348]: MTD_RPMSG ?

File: drivers/mtd/rpmsgmtd.c (hunk 1 above)
Comment (CV-Bowen) [reply drivers/mtd/rpmsgmtd.c]: Done

File: drivers/mtd/rpmsgmtd.c [hunk 3]
```diff
@@ -0,0 +1,1135 @@
...
+ *
+ ****************************************************************************/
+
+static size_t rpmsgmtd_ioctl_arglen(int cmd)
+{
+  switch (cmd)
+    {
+      case MTDIOC_GEOMETRY:
+        return sizeof(struct mtd_geometry_s);
+      case MTDIOC_BULKERASE:
```
[the remark this answers is not in the document]
Comment (CV-Bowen) [reply drivers/mtd/rpmsgmtd.c]: Done

File: drivers/mtd/rpmsgmtd.c [hunk 4]
```diff
@@ -0,0 +1,1135 @@
...
+  msg->request = cmd;
+  msg->arg     = arg;
+  msg->arglen  = arglen;
+
+  if (arglen > 0)
+    {
+      memcpy(msg->buf, (FAR void *)(uintptr_t)arg, arglen);
+    }
+
+  ret = rpmsgmtd_send_recv(priv, RPMSGMTD_IOCTL, false, &msg->header,
```
[the remark this answers is not in the document]
Comment (CV-Bowen) [reply drivers/mtd/rpmsgmtd.c]: Done

File: drivers/mtd/rpmsgmtd.c [hunk 5]
```diff
@@ -0,0 +1,1128 @@
...
+ *
+ * Parameters:
+ *   dev - the mtd device
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+static int rpmsgmtd_get_geometry(FAR struct rpmsgmtd_s *dev)
```
Comment (anchao) [inline drivers/mtd/rpmsgmtd.c:194]: static int rpmsgmtd_get_geometry(FAR struct rpmsgmtd_s *dev)
{ 
  int ret;
    
  ret = nxmutex_lock(&dev->geoexcl);
  if (ret == OK && dev->geo.blocksize == 0)
    {
      /* Get the server mtd device geometry */
          
      ret = rpmsgmtd_ioctl(&dev->mtd, MTDIOC_GEOMETRY,
                           (unsigned long)&dev->geo);
    }

  nxmutex_unlock(&dev->geoexcl);
  return ret;
}

File: drivers/mtd/Kconfig (hunk 2 above)
Comment (CV-Bowen) [reply drivers/mtd/Kconfig]: Done

Comment (CV-Bowen) [reply drivers/mtd/Kconfig]: Both are good for me
MTD_RPMSG: like MTD_PARTITION, implement an abstract layer upper on an exists mtd device, MTD_RPMSG implement an abstract layer upper on a remote mtd device;
RPMSGMTD: like RAMMTD, an independent mtd device implementation.

Comment (CV-Bowen) [reply drivers/mtd/Kconfig]: @xiaoxiang781216 @anchao Should i change it back?

Comment (xiaoxiang781216) [reply drivers/mtd/Kconfig]: Yes, BTW, since the file name is rpmsgmtd.c which is more consistent with RPMSGMTD.

Comment (CV-Bowen) [reply drivers/mtd/Kconfig]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by CV-Bowen** [comment]
CI failed, refresh the defconfig.

**Pull request merged**